libraries/profile.rb

```diff
@@ -6,200 +6,178 @@
-  property :token, [String, nil]
-  property :refresh_token, [String, nil]
-  property :insecure, [TrueClass, FalseClass], default: false
-  property :inspec_version, String, default: 'latest'
-  property :formatter, ['json', 'json-min'], default: 'json-min'
-  property :quiet, [TrueClass, FalseClass], default: true
-  # TODO(sr) it might be nice to default to settings from attributes
-
-  # alternative to (owner, profile)-addressing for profiles,
-  # e.g. for running profiles from disk (coming from some other source)
-  property :path, String
-
-  default_action :execute
-
-  action :fetch do
-    converge_by 'install/update inspec' do
-      chef_gem 'inspec' do
-        version inspec_version if inspec_version != 'latest'
-        compile_time true
-        action :install
-      end
-
-      require 'inspec'
-      # load the supermarket plugin
-      require 'bundles/inspec-supermarket/api'
-      require 'bundles/inspec-supermarket/target'
+class Audit
+  class Resource
+    class ComplianceProfile < Chef::Resource
+      include ComplianceHelpers
+      use_automatic_resource_name
+
+      property :profile, String, name_property: true
+      property :owner, String, required: true
+
+      # to use a chef-compliance server that is used with chef-server integration
+      property :server, [String, URI, nil]
+      property :port, Integer
+      property :token, [String, nil]
+      property :insecure, [TrueClass, FalseClass], default: false
+      property :formatter, ['json', 'json-min'], default: 'json-min'
+      property :quiet, [TrueClass, FalseClass], default: true
+      # TODO(sr) it might be nice to default to settings from attributes
+
+      # alternative to (owner, profile)-addressing for profiles,
+      # e.g. for running profiles from disk (coming from some other source)
+      property :path, String
+
+      default_action :execute
+
+      action :fetch do
+        converge_by 'load required inspec modules' do
+          require 'inspec'
+          # load the supermarket plugin
+          require 'bundles/inspec-supermarket/api'
+          require 'bundles/inspec-supermarket/target'
+
+          # load the compliance api plugin
+          require 'bundles/inspec-compliance/api'
+          require 'bundles/inspec-compliance/http'
+        end
 
-      # load the compliance api plugin
-      require 'bundles/inspec-compliance/api'
-      require 'bundles/inspec-compliance/http'
+        converge_by 'create cache directory' do
+          directory(::File.join(Chef::Config[:file_cache_path], 'compliance')).run_action(:create)
+        end
 
-      check_inspec
-    end
+        converge_by 'fetch compliance profile' do
+          return if path # will be fetched from other source during execute phase
+
+          o, p = normalize_owner_profile
+          Chef::Log.info "Fetch compliance profile #{o}/#{p}"
+
+          path = tar_path
+          access_token = token
+
+          puts "ACCESSTOKEN #{access_token}"
```

Comment:
Being a little nit-picky; line 58 looks to be a left-over debugging artifact.  As-is it is appended to the end of current line in console/logs during converges and doesn't have any context and is empty when `['audit']['collector'] == chef-server`.  Perhaps, if this is valuable/desirable info to have in log files (which I'm not totally convinced it is), then a line like: `Chef::Log.info "Using Access Token: #{access_token}" if access_token` would be more delightful for the users?